### PR TITLE
Badge Razor Components

### PR DESCRIPTION
**Pull request by jonashendrickx**

### CODE REVIEW

File: src/AdminConsole/Components/Shared/Badge.razor [hunk 1]
```diff
@@ -12,7 +14,7 @@
 
     protected override Task OnInitializedAsync()
     {
-        Class = "rounded-md px-2 py-1 text-xs font-semibold";
+        Class = string.Join(' ', BaseClass, ClassVariant, Class);
```
[the remark this answers is not in the document]
Comment (jrmccannon) [reply src/AdminConsole/Components/Shared/Badge.razor:17]: This also changes the order of the old implementation.

File: src/AdminConsole/Pages/Account/Profile.cshtml [hunk 2]
```diff
@@ -8,8 +8,9 @@
     var requestToken = Antiforgery.GetAndStoreTokens(Model.HttpContext).RequestToken;
 }
 
-<panel header="Passkeys">
+<div class="panel">
```
Comment (jrmccannon) [inline src/AdminConsole/Pages/Account/Profile.cshtml:11]: Are we moving away from Panel?

**Review by jrmccannon — COMMENTED** [review]
My questions echo Oleksii's concerns.

File: src/AdminConsole/Pages/App/Settings/Settings.cshtml [hunk 3]
```diff
@@ -16,16 +18,16 @@
                     {
                         if (plan.IsOutdated)
                         {
-                            <badge variant="ColorVariant.Warning" text="New Available"></badge>
+                            <component type="typeof(Badge)" render-mode="Static" param-Variant="@(ColorVariant.Warning)" param-Message="@("New Available")" />
```
[the remark this answers is not in the document]
Comment (jonashendrickx) [reply src/AdminConsole/Pages/App/Settings/Settings.cshtml:21]: This is the razor component, that is the syntax to use it in razor pages

File: src/AdminConsole/Pages/App/Settings/Settings.cshtml [hunk 4]
```diff
@@ -1,11 +1,13 @@
...
+@using Badge = Passwordless.AdminConsole.Components.Shared.Badge
 @model SettingsModel
 
 @{
     ViewData["Title"] = "Settings";
 }
 
-<panel header="Plan">
+<div class="panel">
+    <h2>Plan</h2>
```
[the remark this answers is not in the document]
Comment (jonashendrickx) [reply src/AdminConsole/Pages/App/Settings/Settings.cshtml:10]: razor components do not render inside tag helper

File: src/AdminConsole/Components/Shared/Badge.razor (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonashendrickx) [reply src/AdminConsole/Components/Shared/Badge.razor:17]: The badges use the same base classes for all their variants. Depending on the enum it adds some extra classes for background and text color. Then any other desired changes can be added on top by specifying additional classes.

File: src/AdminConsole/Pages/Account/Profile.cshtml (hunk 2 above)
Comment (jonashendrickx) [reply src/AdminConsole/Pages/Account/Profile.cshtml:11]: You cannot use razor components inside tag helpers

File: src/AdminConsole/Pages/App/Settings/Settings.cshtml (hunk 4 above)
[the remark this answers is not in the document]
Comment (Tyrrrz) [reply src/AdminConsole/Pages/App/Settings/Settings.cshtml:10]: Is `<panel>` a tag helper? Why not make that into a component too then?

File: src/AdminConsole/Pages/App/Settings/Settings.cshtml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Tyrrrz) [reply src/AdminConsole/Pages/App/Settings/Settings.cshtml:21]: That looks horrible 😱 
I thought Razor components could just be rendered as `<Badge ... />`, no?

File: src/AdminConsole/Components/Shared/Badge.razor (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tyrrrz) [reply src/AdminConsole/Components/Shared/Badge.razor:17]: My biggest concern is `Class = string.Join(..., Class)`. The fact that `Class` is set based on itself is weird. I am not fully aware of how the component lifecycle works in Blazor, but where is the `Class` property set originally?

File: src/AdminConsole/Pages/App/Settings/Settings.cshtml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jonashendrickx) [reply src/AdminConsole/Pages/App/Settings/Settings.cshtml:21]: If you use Blazor yes.

**Review by Tyrrrz — APPROVED** [review]

**Review by jrmccannon — APPROVED** [review]

**Pull request merged**